content/docs/user-guide/project-structure/pipelines-files.md

```diff
@@ -0,0 +1,416 @@
+expression can use the `${item.key}` form. Stage names will be appended with a
+zero-based index. For example:
+
+```yaml
+stages:
+  train:
+    foreach:
+      - epochs: 3
+        thresh: 10
+      - epochs: 10
+        thresh: 15
+    do:
+      cmd: python train.py ${item.epochs} ${item.thresh}
+```
+
+```yaml
+# dvc.lock
+stages:
+  train@0:
+    cmd: python train.py 3 10
+  train@1:
+    cmd: python train.py 10 15
+```
+
+DVC can also iterate on a dictionary given directly to `foreach`, resulting in
+two substitution expressions being available: `${key}` and `${item}`. The former
+is used for the stage names:
+
+```yaml
+stages:
+  build:
+    foreach:
+      uk:
+        epochs: 3
+        thresh: 10
+      us:
+        epochs: 10
+        thresh: 15
+    do:
+      cmd: python train.py '${key}' ${item.epochs} ${item.thresh}
+      outs:
+        - model-${key}.hdfs
+```
+
+```yaml
+# dvc.lock
+stages:
+  build@uk:
+    cmd: python train.py 'uk' 3 10
+    outs:
+      - model-uk.hdfs
+  build@us: ...
+```
+
+Importantly, dictionaries [from parameters](#templating) files can be used in
+`foreach` multi-stages as well:
+
+```yaml
+stages:
+  mystages:
+    foreach: ${myobject} # From params.yaml
+    do:
+      cmd: ./script.py ${key} ${item.prop1}
+      outs:
+        - ${item.prop2}
+```
+
+> Note that this feature is not compatible with [templating](#templating) at the
+> moment.
+
+## Specification
+
+| Field            | Description                                                                                                                                                                                                                                                                               |
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `cmd`            | (Required) One or more commands executed by the stage (may contain either a single value or a list). Commands are executed sequentially until all are finished or until one of them fails (see `dvc repro`).                                                                              |
+| `wdir`           | Working directory for the stage command to run in (relative to the file's location). Any paths in other fields are also based on this. It defaults to `.` (the file's location).                                                                                                          |
+| `deps`           | List of <abbr>dependencies</abbr> paths of this stage (relative to `wdir`).                                                                                                                                                                                                               |
```

Comment:
Should `dependencies` be singular (`dependency`)?

Reply:
Nope, it's a list 🙂

Reply:
Wait no, you were right. It should be singular because "paths" is plural 🤦 

Fixed (also in outs)!